### PR TITLE
Don't use std::move

### LINKED ISSUE #53: C++98 compatibility broken
This commit breaks C++98 compatibility by using `std::move` and r-value references: https://github.com/mat007/turtle/pull/41/commits/caef00d2e315cda0697ebf79c4bade888aa9b17d

### PR DESCRIPTION
**Pull request by Flamefire**
This should fix #53 by partially reverting https://github.com/mat007/turtle/pull/41/commits/caef00d2e315cda0697ebf79c4bade888aa9b17d

### CODE REVIEW

File: include/turtle/detail/action.hpp [hunk 1]
```diff
@@ -109,17 +109,11 @@ namespace detail
...
-                    &move< Value >,
+                    &boost::move<Value>,
                     boost::ref( store( boost::move( v ) ) ) ) );
         }
 
     private:
-        template< typename T >
-        static T&& move( T& t )
-        {
-            return std::move( t );
```
[the remark this answers is not in the document]
Comment (Flamefire) [reply include/turtle/detail/action.hpp]: I didn't want to fiddle with the return value too. I'll check CI errors now and see what is required.

File: .travis.yml [hunk 2]
```diff
@@ -69,7 +69,7 @@ script:
     - cd $PROJECT_DIR/build
     - export BOOST_ROOT=$BOOST
     # `--coverage` flags required to generate coverage info for Coveralls
-    - ./build.sh --toolset=$CC "cxxflags=-std=$CXX_STANDARD -Wno-unused-variable -Wno-unused-function -Wno-deprecated-declarations --coverage" "linkflags=--coverage"
+    - ./build.sh --toolset=$CC "cxxflags=-std=$CXX_STANDARD -Wnounused-local-typedef -Wno-unused-variable -Wno-unused-function -Wno-deprecated-declarations --coverage" "linkflags=--coverage"
```
[the remark this answers is not in the document]
Comment (Flamefire) [reply .travis.yml]: Just wrong name: `-Wno-unused-local-typedefs`. Testing locally now

File: include/turtle/detail/action.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Flamefire) [reply include/turtle/detail/action.hpp]: Well, this seems to be valid: https://stackoverflow.com/questions/28595117/why-can-we-use-stdmove-on-a-const-object

Trying a different approach now

[the remark this answers is not in the document]
Comment (mat007) [reply include/turtle/detail/action.hpp]: Well, you can if `f.expect().moves( i );` makes a copy internally and stores it, and it's that copy that gets moved later when triggering the expectation.

[the remark this answers is not in the document]
Comment (mat007) [reply include/turtle/detail/action.hpp]: > Well, this seems to be valid

OK, or that, I don't remember the specifics of what prompted this test case :/

**Comment by mat007** [comment]
Closing as this is included in #59

**Pull request closed without merging**